examples/format_detection_openai_gpt.py

```diff
@@ -0,0 +1,126 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # PII detection with DiscoverX & Azure OpenAI
+# MAGIC
+# MAGIC This notebooks uses [DiscoverX](https://github.com/databrickslabs/discoverx) to run PII detection with [AZURE OpenAI API](https://learn.microsoft.com/en-us/azure/ai-services/openai/chatgpt-quickstart?tabs=command-line&pivots=programming-language-studio) over a set of tables in Unity Catalog.
+# MAGIC
+# MAGIC The notebook will:
+# MAGIC 1. Use DiscoverX to sample a set of tables from Unity Catalog and unpivot all string columns into a long format dataset
+# MAGIC 2. Run format detection with Azure OpenAI GPT model
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Install dependencies
+
+# COMMAND ----------
+
+# MAGIC %pip install openai
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Setup widgets
+
+# COMMAND ----------
+
+dbutils.widgets.text("secret_scope","discoverx","Secret Scope")
+dbutils.widgets.text("open_ai_base","openaibase","Secret Key of Open API Base")
+dbutils.widgets.text("open_ai_key","openaikey","Secret Key of Open AI API Key")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Import required libs and initialize variables
+
+# COMMAND ----------
+
+import openai
+import pandas as pd
+from pyspark.sql.functions import pandas_udf, col, concat, lit, explode, count, avg, min, max, sum,collect_set,concat_ws
+from pyspark.sql.types import ArrayType, StringType, StructType, FloatType, StructField
+from typing import Iterator
+
+# COMMAND ----------
+
+# TODO: Change the table selection
+# from_tables = "sample_data_discoverx.*.*"
+from_tables = "discoverx_sample.sample_datasets.cyber_data"
+
+# TODO: Change the num of rows to sample
+sample_size = 100
+
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Initialize discoverx
+
+# COMMAND ----------
+
+from discoverx import DX
+
+dx = DX()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Transform all sampled tables 
+
+# COMMAND ----------
+
+unpivoted_df = (
+    dx.from_tables(from_tables)
+    .unpivot_string_columns(sample_size=sample_size)
```

Comment:
Note: this only uses string columns, but will ignore arrays, structs, numbers, etc.
We should also build a `unpivot_all_columns_as_string` that casts any column to a string